lib/client.js

```diff
@@ -1331,6 +1331,65 @@ MatrixClient.prototype.invite = function(roomId, userId, callback) {
 };
 
 /**
+ * Invite a user to a room based on their email address.
+ * @param {string} roomId The room to invite the user to.
+ * @param {string} email The email address to invite.
+ * @param {string} displayName The display name to use. If omitted, everything
+ * before the '@' in the email address is used.
+ * @param {module:client.callback} callback Optional.
+ * @return {module:client.Promise} Resolves: TODO
+ * @return {module:http-api.MatrixError} Rejects: with an error response.
+ */
+MatrixClient.prototype.inviteByEmail = function(roomId, email, displayName, callback) {
+    if (utils.isFunction(displayName)) {
+        callback = displayName; displayName = undefined;
+    }
+    if (!displayName) {
+        displayName = email.split("@")[0];
```

Comment:
Oh I see, didn't realise this returned an array of one. This is probably fine then.